packages/ai-chat/src/browser/change-set-file-element.ts

```diff
@@ -214,16 +270,25 @@ export class ChangeSetFileElement implements ChangeSetElement {
     }
 
     onShow(): void {
-        this.changeResource.update({ contents: this.targetState, onSave: content => this.writeChanges(content) });
+        // Ensure we have the latest state when showing
+        if (!this._initialized && this._initializationPromise) {
+            this._initializationPromise.then(() => {
+                // Update the change resource with the correct target state after initialization
+                this.changeResource.update({ contents: this.targetState, onSave: content => this.writeChanges(content) });
+            });
+        } else {
+            this.changeResource.update({ contents: this.targetState, onSave: content => this.writeChanges(content) });
```

Comment:
Do we need to wait for initialization to handle anything related to `changeResource`?

Reply:
Unfortunately yes, because `writeChanges` triggers the file system listener from `listenForOriginalFileChanges`, which uses the original content to determine the state of the element. I moved the check directly to listener now, I think it makes more sense to have it there directly.